### PR TITLE
refactor(server): calculate asset type server side

### PR DESCRIPTION
**Pull request by jrasm91**
In this PR:
- Remove `assetType` from asset create dto
- Remove `fileExtension` from asset create dto
- Remove mime type from upload request

Tested Scenarios:
- Upload video to web (.mp4, .mov)
- Upload image to web (.cr2, .raw, .arw, .jpg, .heic, .nef)

TODO:
- Mobile changes

Not Tested:
- Upload from mobile

### CODE REVIEW

**Comment by uhthomas** [comment]
:smile: 
image

File: web/src/lib/utils/file-uploader.ts [hunk 1]
```diff
@@ -132,19 +127,15 @@ async function fileUploader(
     // Get asset duration
     formData.append('duration', '0:00:00.000000');
 
-    // Get asset file extension
-    formData.append('fileExtension', '.' + fileExtension);
-
     // Get asset binary data with a custom MIME type, because browsers will
     // use application/octet-stream for unsupported MIME types, leading to
     // failed uploads.
```
Comment (uhthomas) [inline web/src/lib/utils/file-uploader.ts:132]: This comment is out of date.

**Comment by uhthomas** [comment]
Fixes #2853.

**Comment by uhthomas** [comment]
Related #2824.

**Comment by uhthomas** [comment]
Hope you don't mind, I've marked this as a draft PR as there are still outstanding changes (i.e mobile). Wouldn't want it to be accidentally merged!

**Comment by jrasm91** [comment]
@alextran1502 or @fyfrey - can you help with the mobile changes? Just need to delete a few things:
-  `assetType` and `fileExtension` in `backup.service.dart` (create asset request)
- `contentType` also no longer needs to be sent in the create asset request
- I believe the entire `files_helper.dart` can also be removed.

Thanks in advance.

**Review by alextran1502 — APPROVED** [review]
Tested upload on mobile for
- photos
- videos
- LivePhotos
- Profile Picture

**Pull request merged**